Code/autopkglib/SparkleUpdateInfoProvider.py

```diff
@@ -118,27 +125,88 @@ class SparkleUpdateInfoProvider(Processor):
         },
     }
 
+    def prepare_curl_cmd(self, url, headers=None):
+        """Assemble curl command and return it."""
+
+        curl_cmd = [
+            super(SparkleUpdateInfoProvider, self).curl_binary(),
+            "--location",
+            "--compressed",
+        ]
+        super(SparkleUpdateInfoProvider, self).add_curl_common_opts(curl_cmd)
+        if headers:
+            for header, value in headers.items():
+                curl_cmd.extend(["--header", "%s: %s" % (header, value)])
+        curl_cmd.append(url)
+        return curl_cmd
+
     def fetch_content(self, url, headers=None):
         """Returns content retrieved by curl, given a url and an optional
         dictionary of header-name/value mappings. Logic here borrowed from
         URLTextSearcher processor."""
 
+        curl_cmd = self.prepare_curl_cmd(url, headers)
         try:
-            cmd = [self.env["CURL_PATH"], "--location", "--compressed"]
-            if headers:
-                for header, value in headers.items():
-                    cmd.extend(["--header", "%s: %s" % (header, value)])
-            cmd.append(url)
-            proc = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-            (data, stderr) = proc.communicate()
-            if proc.returncode:
-                raise ProcessorError("Could not retrieve URL %s: %s" % (url, stderr))
-        except OSError:
+            content = super(SparkleUpdateInfoProvider, self).download(curl_cmd)
+        except ProcessorError:
             raise ProcessorError("Could not retrieve URL: %s" % url)
```

Comment:
Why bother catching and then raising the same exception? The download() function already raises an exception that reports a curl failure. I don't really see the benefit of catching it just to reraise with different phrasing.

Reply:
Ok. I remove the code in 94022d3.